All metadata should be in the metadata step of the add event wizard

lkiesow opened the pull request:
The “source” step of the add event wizard also contains metadata. This doesn't really make sense since there is a whole step dedicated to metadata. We should move “Start date” in there.

- The problem: This field should only be shown for a file upload. Not for a scheduled event.
- The solution: Make “Source” step 1 and metadata step 2. We then know if we need a start date once we are in the metadata tab.

This fixes #487

lkiesow committed:
All metadata should be in the metadata step of the add event wizard

The “source” step of the add event wizard also contains metadata. This
doesn't really make sense since there is a whole step dedicated to
metadata. We should move “Start date” in there.

- The problem: This field should only be shown for a file upload. Not for a scheduled event.
- The solution: Make “Source” step 1 and metadata step 2. We then know
  if we need a start date once we are in the metadata tab.

This fixes #487

JulianKniephoff committed:
"Restore" ESLint support

I removed our leftover config in #700 with a promise to bring ESLint back.
Well this is it, for now!

This introduces an ESLint config file in the new flat file format,
which will be required for ESLint `>= 9`. It is already supported in version 8, though,
which we are still stuck on because of the (unmaintained) config we are/were using.
I briefly tried to port this project over to `@opencast/eslint-config-ts-react`,
but the code said "no," basically. ;P

This has a few unfortunate consequences. See #713. I plan to address these in due time, though.

Note that this does **not** introduce automatic/live checking of the code
during development. I only added a call to the linter to the build command,
in turn making CI fail when there are warnings.
I created #712 to track the live thing if we want that.

Commit:
Merge branch 'opencast:main' into issue-640

JulianKniephoff committed:
Get rid of Vite build warnings

There are quite a few errors like the following in the output of `npm run build`:

    Error when using sourcemap for reporting an error: Can't resolve original location of error.

They all relate to the `@mui` package(-s). This seems to be related to https://github.com/vitejs/vite/issues/15012.
I don't fully understand it, but apparently Rollup, which is used by Vite internally somehow,
doesn't preserve enough information to generate source maps properly.

This patch adds a plugin to do exactly that. We should probably remove it in the long run,
if the upstream issue(-s) ever get fixed. I created #708 to track that.

/cc https://github.com/opencast/opencast-admin-interface/pull/685

rlucke committed:
remove outline: none for a11y reasons

lkiesow committed:
Merge pull request #703 from Arnei/fix-capture-agent-table-pagination

Fix pagination in capture agent table

Arnei committed:
Typing event details

Adds typescript types to the parameters of
the `EventDetails` component and various
related components.

Arnei committed:
Actually set initial sort column to "date" instead

Solves the same problem as the previous commit,
but instead of setting "title" as the default, this
instead sets "date" as the default.

Arnei committed:
Merge pull request #664 from rlucke/issue-640

increase contrast ratio in wizard modal

Arnei committed:
Merge pull request #700 from JulianKniephoff/remove-eslint-config

Remove unused `eslint` config

lkiesow committed:
Fix type error for missing organization property

If an organization property does not exist, the admin interface
completely breaks in some cases. This seems to happen on
admin-interface.opencast.org right now:

```
Uncaught TypeError: u.toLowerCase is not a function
    qr Footer.tsx:25
    React 8
    D scheduler.production.min.js:13
    $ scheduler.production.min.js:14
Footer.tsx:25:80
```

What's interesting is that this only seems to happen with the compiled
code. Starting the development server, calling `toLowerCase()` just
works.

In any case, this patch fixes the issue by providing the default value
first, so that `toLowerCase()` is always called on a string.

lkiesow committed:
Merge pull request #704 from Arnei/make-selected-filter-box-smaller

Make selected table filter smaller

lkiesow committed:
Merge pull request #702 from Arnei/refetch-acl-after-save

Refetch event ACL after saving

Commit:
Merge branch 'opencast:main' into issue-617

Commit:
Merge branch 'opencast:main' into issue-624

rlucke committed:
fix a11y issues in header nav

lkiesow committed:
Merge pull request #701 from Arnei/fix-date-table-filter-from-column

Fix date table filter from event column

Commit:
Merge branch 'opencast:main' into issue-605

Commit:
Merge branch 'opencast:main' into issue-617

Commit:
Merge branch 'opencast:main' into issue-624

Commit:
Allow admins to save ACLs with no write roles

rlucke committed:
fix #603

Commit:
Disable ACL save button

Arnei committed:
Set default sort order to DESC

Sets the default sort order to DESC, so that when
we initially sort the table by date, the "newest"
events are listed first. Requires that we set ASC
explicitly for all other tables.

Commit:
Update Table.tsx

use button instead of div

Arnei committed:
Merge pull request #707 from JulianKniephoff/npm-audit-fix

`npm audit fix`

lkiesow committed:
Merge pull request #699 from Arnei/sort-by-title-asap

Set initial sort column to "date"

rlucke committed:
remove outline:0

rlucke committed:
wrap icons in button tag

Commit:
Merge branch 'opencast:main' into issue-617

Commit:
Merge branch 'opencast:main' into issue-622

Arnei committed:
Merge remote-tracking branch 'origin/main' into to-redux-toolkit-table

mtneug commented:
Two other fields missing are "location" and "duration" (used for calculating the end date in the Admin UI). I set this for file uploads if I know these files are from a recording (e.g. CA did the recording, but editing was done locally). Is this in the scope of this PR, or should I create another issue?

Commit:
Merge branch 'opencast:main' into issue-616

rlucke committed:
add toggle button to main menu

rlucke committed:
add tooltip to main menu

Arnei committed:
Default typing for createAsyncThunk

Wraps `createAsyncThunk` in `createAsyncThunk.withTypes`, allowing us to
provide default typing and remove certain type
assertions.

Commit:
Custom roles in ACL policy

Commit:
Don't show unsaved changes warning after saving

rlucke committed:
remove tabIndex from DropDown

Commit:
Add keys to remove focus on input fields

rlucke committed:
add live area and turn divs into buttons

lkiesow commented:
You mean, you want these two additional fields in the recording metadata section? I think that's unrelated enough to not wait for that when merging this pull request. Nevertheless, I'll try taking a peek at the issue since I now kind of know the code and I hope it's not too hard to do.

wsmirnow committed:
Hide type and flavor of add asset items

To be consistent, the Add-Asset dialog shouldn't show the type and flavor of the items, same as Add-event dialog.

wsmirnow committed:
New event summary should also hide asset type and flavor

Commit:
Add sub navigation tabs in event details assets

Commit:
Hide tooltip on hover

Commit:
Show main navigation tooltips on the right

lkiesow committed:
Merge pull request #709 from JulianKniephoff/build-warnings

Get rid of Vite build warnings

lkiesow committed:
Merge pull request #720 from dennis531/acl-save-button

Disable ACL save button

lkiesow committed:
Merge pull request #730 from dennis531/save-dialog

Don't show unsaved changes warning after saving

lkiesow committed:
Merge pull request #740 from dennis531/tooltip-covering

Hide tooltip on hover

lkiesow committed:
Merge pull request #739 from dennis531/assets-nav

Add tabs in event details assets

mtneug commented:
> You mean, you want these two additional fields in the recording metadata section? I think that's unrelated enough to not wait for that when merging this pull request. Nevertheless, I'll try taking a peek at the issue since I now kind of know the code and I hope it's not too hard to do.

Fair. I created issue #742 to track this.

Arnei committed:
Merge pull request #714 from JulianKniephoff/eslint

"Restore" ESLint support

Arnei committed:
Merge pull request #705 from JulianKniephoff/nitpicks

Remove poorly formatted dead code

Arnei committed:
Merge pull request #710 from rlucke/issue-617

Table focus is not shown #617

Arnei committed:
Merge pull request #715 from lkiesow/tolowercase-not-a-function

Fix type error for missing organization property

Arnei committed:
Merge remote-tracking branch 'origin/main' into typing-event-details

Arnei committed:
Fix checkstyle issues and siable sourcemaps

ESLint is back, yay. But while it was gone, other
PRs introduced checkstyle issues, oh no. This fixes
those issues.

Secondly, this also disables sourcemaps. I was unable
to build the project due to out-of-memory errors,
and disabling sourcemaps proved a workaround.
We can probably enable them again once we get
rid of all the old dependencies that come with our
current ESLint config.

Commit:
Merge branch 'opencast:main' into issue-625

Commit:
Merge branch 'opencast:main' into issue-598

Commit:
Merge branch 'opencast:main' into issue-615

Arnei committed:
Merge pull request #718 from rlucke/issue-605

fix a11y issues in header nav

Arnei committed:
Actually do not disable sourcemaps

Arnei committed:
Merge pull request #721 from rlucke/issue-622

 Pagination selector cannot be controlled with the keyboard #622

Arnei committed:
Merge pull request #724 from rlucke/issue-616

 Table filters cannot be reached with the keyboard #616

Arnei committed:
Merge pull request #731 from rlucke/issue-598

remove tabindex from drop downs

Arnei committed:
Merge pull request #727 from rlucke/issue-615

 Hamburger menu cannot be reached with the keyboard #615

Arnei committed:
Merge pull request #737 from wsmirnow/add-asset-items-format

Hide type and flavor of add asset items

Arnei committed:
Merge pull request #736 from rlucke/issue-625

 Einbettungscode Dialog a11y issues #625

Arnei committed:
Merge pull request #671 from rlucke/issue-624

Use button tag for stat filter items

JulianKniephoff committed:
Merge remote-tracking branch 'origin/main' into fix-checkstyle

JulianKniephoff committed:
Fix more lints

Arnei committed:
Don't word wrap on table headers

We're currently applying arbitrary word-wrapping
to our table cells to keep very long strings from
overflowing. This can cause some columns to
get very little width in certain cases. This removes
word-wrapping from table headers to rectify that
issue. Table header contents are controlled by us
and should not be terribly long either way, so in
my opinion this is fine.

Arnei committed:
Merge pull request #743 from Arnei/fix-checkstyle

Fix checkstyle issues and disable sourcemaps

Arnei committed:
Merge remote-tracking branch 'origin/main' into make-table-header-not-wrap

Arnei committed:
Merge remote-tracking branch 'origin/main' into typing-event-details

Arnei committed:
Prevent icon in table header from line breaking

Applies flexbox to table headers to keep table
header contents from line breaking. This is
supposed to help with cases where the title
would be on one line and the sort icon on the next,
wasting space and looking quite bad.

Arnei committed:
Remove unused import

Arnei committed:
Add missing translation key

Translation key for the button to studio was missing.

JulianKniephoff committed:
Add a very restricted form of `no-tabs` to the ESLint config

This is in preparation to adopting `@opencast/eslint-config-ts-react`,
which includes `no-tabs`.

We use tabs for indentation in some places, but also for other things.
The indentation tabs are easy to get rid of automatically, if a bit annoying.
It is the **other** instances that are not so easily picked out,
and especially distinguished from valid tab usages (think literals).

So I'm getting these out of the way first.

JulianKniephoff committed:
Remove some dead code

wsmirnow committed:
Asset-Upload fixed

This patch fixes the asset upload.

wsmirnow committed:
Fixed asset upload on new event

lkiesow committed:
Main menu tooltip position

This patch adjusts the placement of the main menu tooltip to appear only
atop the icon and be placed to the right like with all other main menu
tooltips.

lkiesow committed:
Improve embed codes

This patch improves the embed code which was somewhat wird. For example,
`style="border:0px #FFFFFF none;"` did not make much sense since the
syntax for `border` is `width | style | color`¹.

[[1]](https://developer.mozilla.org/en-US/docs/Web/CSS/border)

Arnei committed:
Deselect all rows after start task

After starting a task on the selected events,
this causes all events to become unselected.

lkiesow committed:
Remove Tooltip from Stats Filter

This patch removes the tooltips from the stats filters since they always
just showed exactly what already was displayed on the label without
tooltip. Additional visual clutter with no additional information does
not make much sense.

Commit:
Automatically update translation keys

Arnei committed:
Fix indentation

Arnei committed:
"Unsaved changes" popup showing up but changes are saved

Fix an issue where a browser popup informing
the user about unsaved changes would show up
when trying to close the series details modal after
editing acls, no matter if they actually saved their
changes or not.

Arnei committed:
Merge pull request #711 from Arnei/typing-event-details

Typing event details

Arnei committed:
Merge pull request #728 from Arnei/default-typing-for-createAsyncThunk

Default typing for createAsyncThunk

Arnei committed:
Use "presentation" instead of "slides" for upload

Changes the wording for the source upload in the
create event dialog to use the word "presentation"
instead of "slides".

Arnei committed:
Merge remote-tracking branch 'origin/main' into to-redux-toolkit-table

Arnei committed:
Merge remote-tracking branch 'origin/main' into fix-unsaved-changes-series-acl

Arnei committed:
Add typing to wizard stepper

More specifically to wizardUtils

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin from 7.13.1 to 7.14.1

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 7.13.1 to 7.14.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v7.14.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @types/node from 20.14.1 to 20.14.9

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.14.1 to 20.14.9.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @eslint/js from 9.5.0 to 9.6.0

Bumps [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js) from 9.5.0 to 9.6.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/commits/v9.6.0/packages/js)

---
updated-dependencies:
- dependency-name: "@eslint/js"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @mui/material from 5.15.19 to 5.15.21

Bumps [@mui/material](https://github.com/mui/material-ui/tree/HEAD/packages/mui-material) from 5.15.19 to 5.15.21.
- [Release notes](https://github.com/mui/material-ui/releases)
- [Changelog](https://github.com/mui/material-ui/blob/v5.15.21/CHANGELOG.md)
- [Commits](https://github.com/mui/material-ui/commits/v5.15.21/packages/mui-material)

---
updated-dependencies:
- dependency-name: "@mui/material"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump react-datepicker from 6.9.0 to 7.2.0

Bumps [react-datepicker](https://github.com/Hacker0x01/react-datepicker) from 6.9.0 to 7.2.0.
- [Release notes](https://github.com/Hacker0x01/react-datepicker/releases)
- [Commits](https://github.com/Hacker0x01/react-datepicker/compare/v6.9.0...v7.2.0)

---
updated-dependencies:
- dependency-name: react-datepicker
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump docker/build-push-action from 5 to 6

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 5 to 6.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/v5...v6)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump softprops/turnstyle from 1 to 2

Bumps [softprops/turnstyle](https://github.com/softprops/turnstyle) from 1 to 2.
- [Release notes](https://github.com/softprops/turnstyle/releases)
- [Changelog](https://github.com/softprops/turnstyle/blob/master/CHANGELOG.md)
- [Commits](https://github.com/softprops/turnstyle/compare/v1...v2)

---
updated-dependencies:
- dependency-name: softprops/turnstyle
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Arnei committed:
Add typing to dateUtils.ts

Adds typescript to the methods in dateUtils.ts.

Arnei committed:
Add typing to componentStyles.ts

Adds typescript to componentStyles.ts.

lkiesow committed:
Merge pull request #760 from opencast/dependabot/npm_and_yarn/typescript-eslint/eslint-plugin-7.14.1

Bump @typescript-eslint/eslint-plugin from 7.13.1 to 7.14.1

lkiesow committed:
Merge pull request #763 from opencast/dependabot/npm_and_yarn/types/node-20.14.9

Bump @types/node from 20.14.1 to 20.14.9

lkiesow committed:
Merge pull request #764 from opencast/dependabot/npm_and_yarn/eslint/js-9.6.0

Bump @eslint/js from 9.5.0 to 9.6.0

dependabot[bot] committed:
Bump typescript from 5.4.5 to 5.5.2

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 5.4.5 to 5.5.2.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Changelog](https://github.com/microsoft/TypeScript/blob/main/azure-pipelines.release.yml)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v5.4.5...v5.5.2)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @types/lodash from 4.17.4 to 4.17.6

Bumps [@types/lodash](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/lodash) from 4.17.4 to 4.17.6.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/lodash)

---
updated-dependencies:
- dependency-name: "@types/lodash"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

lkiesow committed:
Merge pull request #774 from opencast/dependabot/github_actions/docker/build-push-action-6

Bump docker/build-push-action from 5 to 6

lkiesow committed:
Merge pull request #767 from opencast/dependabot/npm_and_yarn/typescript-5.5.2

Bump typescript from 5.4.5 to 5.5.2

lkiesow committed:
Merge pull request #772 from opencast/dependabot/npm_and_yarn/types/lodash-4.17.6

Bump @types/lodash from 4.17.4 to 4.17.6

lkiesow committed:
Merge pull request #766 from opencast/dependabot/npm_and_yarn/mui/material-5.15.21

Bump @mui/material from 5.15.19 to 5.15.21

lkiesow committed:
Merge pull request #768 from opencast/dependabot/npm_and_yarn/react-datepicker-7.2.0

Bump react-datepicker from 6.9.0 to 7.2.0

dependabot[bot] committed:
Bump sass from 1.77.4 to 1.77.6

Bumps [sass](https://github.com/sass/dart-sass) from 1.77.4 to 1.77.6.
- [Release notes](https://github.com/sass/dart-sass/releases)
- [Changelog](https://github.com/sass/dart-sass/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sass/dart-sass/compare/1.77.4...1.77.6)

---
updated-dependencies:
- dependency-name: sass
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

lkiesow committed:
Merge pull request #762 from opencast/dependabot/npm_and_yarn/sass-1.77.6

Bump sass from 1.77.4 to 1.77.6

dependabot[bot] committed:
Bump vite from 5.2.12 to 5.3.2

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.2.12 to 5.3.2.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.3.2/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @reduxjs/toolkit from 2.2.5 to 2.2.6

Bumps [@reduxjs/toolkit](https://github.com/reduxjs/redux-toolkit) from 2.2.5 to 2.2.6.
- [Release notes](https://github.com/reduxjs/redux-toolkit/releases)
- [Commits](https://github.com/reduxjs/redux-toolkit/compare/v2.2.5...v2.2.6)

---
updated-dependencies:
- dependency-name: "@reduxjs/toolkit"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Arnei committed:
Add typing to bulkActionUtils.ts

Add typescript to the utility functions in
bulkactionUtils.ts

Commit:
Prevent overflow by using flexbox in header nav bar

Arnei committed:
Add typing to RegistrationModal

Adds typesript to the registration modal. Not perfect
as I could not quite figure out how to properly
work with literal type unions, but should prove
helpful to whoever will be fixing the registration
modal in the future nonetheless.

Arnei committed:
Add typing to AclUtils.ts

Add typescript to aclUtils.ts

lkiesow committed:
Merge pull request #769 from opencast/dependabot/npm_and_yarn/vite-5.3.2

Bump vite from 5.2.12 to 5.3.2

dependabot[bot] committed:
Bump uuid and @types/uuid

Bumps [uuid](https://github.com/uuidjs/uuid) and [@types/uuid](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/uuid). These dependencies needed to be updated together.

Updates `uuid` from 9.0.1 to 10.0.0
- [Changelog](https://github.com/uuidjs/uuid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/uuidjs/uuid/compare/v9.0.1...v10.0.0)

Updates `@types/uuid` from 9.0.8 to 10.0.0
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/uuid)

---
updated-dependencies:
- dependency-name: uuid
  dependency-type: direct:development
  update-type: version-update:semver-major
- dependency-name: "@types/uuid"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Fix merge conflicts and warnings

Commit:
Add more descriptive handler comment

Arnei committed:
Add typing to user details

Add typescript to various user details components.

Arnei committed:
Add typing to group details

Add typescript to group details and assorted
components. Although a lot of typing was present
already, this just finishes up on the few missing
parts.

Commit:
Hide file remove button if no record file selected

Arnei committed:
Add typing to acl details

Add typescript to acl details and assorted
components. Although a lot of typing was
present already, this just finishes up on the few missing parts.

Arnei committed:
Add typing to ModalNavigation

Adds typescript to the ModalNavigation component.

Arnei committed:
Add typing to BarChart

Add typescript to the BarChart component.

lkiesow committed:
Merge pull request #719 from dennis531/admin-no-write-roles

Allow admins to save ACLs with no write roles

lkiesow committed:
Merge pull request #746 from Arnei/prevent-sort-icon-from-breaking-table-headers

Prevent sort icon from breaking table headers

lkiesow committed:
Merge pull request #747 from Arnei/missing-translation-keys-studio

Add missing translation key

lkiesow committed:
Merge pull request #754 from Arnei/deslect-rows-after-start-task

Deselect all rows after start task

lkiesow committed:
Merge pull request #757 from Arnei/fix-unsaved-changes-series-acl

"Unsaved changes" popup showing up but changes are saved

lkiesow committed:
Merge pull request #758 from Arnei/redo-wording-for-presentation-upload

Use "presentation" instead of "slides" for upload

lkiesow committed:
Merge pull request #780 from dennis531/nav-flex

Prevent overflow by using flexbox in header nav bar

lkiesow committed:
Merge pull request #782 from Arnei/typing-aclUtils

Add typing to AclUtils.ts

lkiesow committed:
Merge pull request #748 from JulianKniephoff/dead-code

Remove some dead code

Commit:
Automatically update translation keys

lkiesow committed:
Merge pull request #788 from Arnei/typing-barchart

Add typing to BarChart

lkiesow committed:
Merge pull request #787 from dennis531/file-remove

Hide file remove button if no record file selected

lkiesow committed:
Replace title attribute in footer elements with tooltips

The footer still used `title` attributes instead of the tooltips we use
everywhere else. This patch fixes that.

Arnei committed:
Merge remote-tracking branch 'origin/main' into to-redux-toolkit-table

Arnei committed:
Merge remote-tracking branch 'origin/main' into typing-acldetails

Arnei committed:
Add typing to ConfirmModal

Add typescript to the ConfirmModal component.

Arnei committed:
Add typing to NewResourceModal

Add typescript to the NewResourceModal component.

Arnei committed:
Add typing to RecordingDetailsModal

Add typescript to the RecodringDetailsModal component.

Arnei committed:
Merge pull request #751 from lkiesow/main-menu-tooltip

Main menu tooltip position

Arnei committed:
Merge pull request #761 from opencast/dependabot/npm_and_yarn/reduxjs/toolkit-2.2.6

Bump @reduxjs/toolkit from 2.2.5 to 2.2.6

dependabot[bot] committed:
Bump @mui/x-date-pickers from 7.6.1 to 7.8.0

Bumps [@mui/x-date-pickers](https://github.com/mui/mui-x/tree/HEAD/packages/x-date-pickers) from 7.6.1 to 7.8.0.
- [Release notes](https://github.com/mui/mui-x/releases)
- [Changelog](https://github.com/mui/mui-x/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mui/mui-x/commits/v7.8.0/packages/x-date-pickers)

---
updated-dependencies:
- dependency-name: "@mui/x-date-pickers"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Arnei committed:
Merge pull request #765 from opencast/dependabot/npm_and_yarn/mui/x-date-pickers-7.8.0

Bump @mui/x-date-pickers from 7.6.1 to 7.8.0

dependabot[bot] committed:
Bump react-router-dom from 6.23.1 to 6.24.0

Bumps [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom) from 6.23.1 to 6.24.0.
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router-dom/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router-dom@6.24.0/packages/react-router-dom)

---
updated-dependencies:
- dependency-name: react-router-dom
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Arnei committed:
Add typing to validate.ts

Fix typescript errors in validates.ts

Commit:
Fix empty presenters filter field in events

lkiesow committed:
Fix processing status translation key

This patch fixes the translation key for “Processing canceled” so that
the status will actually be translated.

Arnei committed:
Fix unique key prop

No idea if this was actually causing errors, but this
fixes the console warning at least.

Arnei committed:
Fixes video preview displaying wrong video

The video preview player in the event assets details
would try to display the media from the assets
previously visited, which would lead to the
player displaying no or the wrong video. This fixes
that by ensuring that the player reloads when the
url changes.

Arnei committed:
Port password strength indicator

Reintroduces the password strenght indicator
from the old admin ui by porting the code from
the original PR: https://github.com/opencast/opencast/pull/3343

Fixes #72.

Arnei committed:
Merge pull request #777 from Arnei/type-dateUtils

Add typing to dateUtils.ts

Arnei committed:
Merge pull request #778 from Arnei/typing-componentStyles

Add typing to componentStyles.ts

Arnei committed:
Merge pull request #779 from Arnei/typing-bulkActionutils

Add typing to bulkActionUtils.ts

Arnei committed:
Merge pull request #781 from Arnei/typing-registration-modal

Add typing to RegistrationModal

Arnei committed:
Merge pull request #783 from Arnei/typing-users

Add typing to user details

Arnei committed:
Merge pull request #771 from opencast/dependabot/npm_and_yarn/multi-87b2ad6fb3

Bump uuid and @types/uuid

Arnei committed:
Merge pull request #770 from opencast/dependabot/npm_and_yarn/react-router-dom-6.24.0

Bump react-router-dom from 6.23.1 to 6.24.0

Arnei committed:
Merge pull request #759 from Arnei/wizard-stepper-typing

Add typing to wizard stepper

Arnei committed:
Merge pull request #755 from lkiesow/stats-tooltip

Remove Tooltip from Stats Filter

Arnei committed:
Merge branch 'main' into typing-groupdetails

Arnei committed:
Merge pull request #785 from Arnei/typing-acldetails

Add typing to acl details

Arnei committed:
Merge pull request #786 from Arnei/typing-modal-navigation

Add typing to ModalNavigation

Arnei committed:
Merge pull request #793 from lkiesow/footer-tooltips

Replace title attribute in footer elements with tooltips

Arnei committed:
Merge pull request #795 from Arnei/typing-confirmmodal

Add typing to ConfirmModal

Arnei committed:
Merge pull request #796 from Arnei/typing-confirmationmodal

Add typing to NewResourceModal

Arnei committed:
Merge pull request #797 from Arnei/typing-recordingdetailsmodal

Add typing to RecordingDetailsModal

Arnei committed:
Merge pull request #798 from Arnei/typing-validate

Add typing to validate.ts

Arnei committed:
Merge pull request #784 from Arnei/typing-groupdetails

Add typing to group details

Arnei committed:
Merge remote-tracking branch 'origin/main' into to-redux-toolkit-table

Arnei committed:
Switch green and blue in password strenght indicator

Commit:
Show workflow operations when clicking on status in events table

Open the workflow operations of the last workflow.

Commit:
Merge select and create policy dropdowns

Changes:
- Dropdowns for new policies are always creatable
- Allow creating new roles in existing policies

Commit:
Fix merge conflicts

Commit:
Fix style of saved filter sets

Changes:
- Fix class name
- Use flexbox
- Allow arbitrary word breaks

Commit:
Lower maximum z-index than tooltip index

JulianKniephoff committed:
Fix a broken CSS class

No pun intended.

Commit:
Fix tooltip title of saved filter

JulianKniephoff committed:
Merge branch 'main' into tabs

May contain further stylistic nitpicks

Commit:
Show cursor pointer on header bar icons

wsmirnow committed:
Update src/i18n/org/opencastproject/adminui/languages/lang-en_US.json

Co-authored-by: Arne Wilken <Arnei@users.noreply.github.com>

wsmirnow committed:
Suggested PR changes

Commit:
Sort filters alphabetically

Commit:
Add check mark to selected language

Commit:
Close tooltip when header menu is open

Commit:
Fix indentation

Commit:
Update tooltip of event table state

Commit:
Fix show workflows overview if no workflows available

Commit:
Open scheduling tab for scheduled and recording events

Commit:
Remove data fetch logic from event status cell

Commit:
Fix eslint warnings

lkiesow committed:
Merge pull request #799 from dennis531/presenter-filter

Fix empty presenters filter field in events

lkiesow committed:
Merge pull request #815 from Arnei/password-strenght-indicator

Port password strength indicator

lkiesow committed:
Merge pull request #818 from dennis531/saved-filters-style

Fix style of saved filter sets

lkiesow committed:
Merge pull request #819 from dennis531/fix-tooltip-overlap

Set lower maximum z-index than tooltip index

lkiesow committed:
Merge pull request #820 from dennis531/saved-filter-tooltip

Fix tooltip title of saved filter

lkiesow committed:
Merge pull request #821 from dennis531/icon-pointer

Show cursor pointer on header bar icons

lkiesow committed:
Merge pull request #822 from dennis531/filter-sorted

Sort filters alphabetically

lkiesow committed:
Merge pull request #823 from dennis531/language-indicator

Add check mark to selected language

lkiesow committed:
Merge pull request #813 from Arnei/fix-unique-key

Fix unique key prop

lkiesow committed:
Merge pull request #775 from opencast/dependabot/github_actions/softprops/turnstyle-2

Bump softprops/turnstyle from 1 to 2

Arnei committed:
Don't forget to add mimetype to source element

Removed the hardcoded mimetype in an earlier
commit, but forgot to add the dynamic mimetype.
Whoops!

Arnei committed:
Merge pull request #750 from wsmirnow/fix-asset-upload

Asset-Upload fixed

Arnei committed:
Merge pull request #801 from lkiesow/processing-canceled

Fix processing status translation key

Commit:
Move workflow id selector into sub tabs

Arnei commented on src/components/events/partials/ModalTabsAndPages/NewSourcePage.tsx:
```diff
@@ -84,6 +81,7 @@ const NewSourcePage = <T extends RequiredFormProps>({
 		dispatch(fetchRecordings("inputs"));
 
 		// validate form because dependent default values need to be checked
+    console.info(formik);
```
Leftover code

Arnei commented on src/components/events/partials/ModalTabsAndPages/NewSeriesMetadataPage.tsx:
```diff
@@ -0,0 +1,80 @@
+/**
+ * This component renders the metadata page for new events and series in the wizards.
+ */
+const NewSeriesMetadataPage = <T,>({
```
This is a lot of code duplication just because we want to add start date to the event metadata tab. I would prefer if we could avoid this.

lkiesow replied:
I've tried that first, but there was more special handling between both since not only the metadata is different, but also the control elements. At the ed it seemed to complicated to me and I gave up. So… yes, I at least thought  about this.

Arnei reviewed (changes requested):
Works. But I'm not totally happy with it ^^'.

Does the start date field require special attention? If not, I would prefer if start date was grouped together with the other metadata. Having it in its own special box might set wrong expectations, i.e. "Is 'recording metadata' a different metadata catalog from 'metadata'?".
![Bildschirmfoto vom 2024-07-11 08-57-11](https://github.com/opencast/opencast-admin-interface/assets/14070005/d083838d-5930-4d65-b0c3-933a3fdd94a6)

JulianKniephoff committed:
Merge remote-tracking branch 'origin/main' into tabs

lkiesow committed:
Merge Metadata

This patch removes the special recording metadata section and merges it
into the default metadata to avoid the appearance that these are
separate metadata catalogs.

lkiesow commented:
Fixed the minor issues and merged the different metadata sections. Now they shouldn't seem like different catalogs any longer.

Arnei commented:
> Now they shouldn't seem like different catalogs any longer.

Can confirm, looks good to me now.

I *was* going to complain that we should need extra code for rendering start_date, as it is part of the standard metadata catalog and therefore should be able to be rendered just like any other metadata field. But it turns the reason we cannot do that is ultimately the backends fault, as the relevant endpoint (admin-ng/event/new/metadata) explicitly excludes start_date (among other fields). Whether or not this is reasonable likely requires additional consideration, so I'm ok with leaving this as it is for now.

Arnei reviewed (changes requested):
If you schedule a new event and have asset upload configured, the "Previous" button on the metadata tab does nothing. It should bring me back to the "Source" tab.

Commit:
Fix missing key in metadata translations

Commit:
Remove unused imports

Commit:
Add translation for no options message in dropdown

Commit:
Fix height of filters

Basically adds flexbox to elements

Arnei committed:
Merge pull request #729 from dennis531/custom-roles

Custom roles in ACL policy

Arnei committed:
Merge pull request #733 from dennis531/input-field-blur

Add keyboard keys to remove focus on input fields

lkiesow commented:
Rebased to resolve merge conflicts and fixed the next/previous problem.

Commit:
Fix recursion in effect when fetching series acls

dennis531 committed:
Add gap to filters

Co-authored-by: Lars Kiesow <lkiesow@uos.de>

Commit:
Automatically update translation keys

Arnei reviewed (changes requested):
Can confirm, it is fixed.

Did find another small issue while testing though, sorry :)
The validation for the main metadata is not happening anymore. Effectively this means you could create an event without a title.

Commit:
Remove eslint comment

gregorydlogan committed:
Merge branch 'to-redux-toolkit-table' of Arnei/opencast-admin-interface into main

Pull request #695

  Modernize redux: tableSlice

Commit:
Remove div container around select component

lkiesow committed:
Merge pull request #841 from dennis531/fix-dropdown-display

Remove div container around select component

lkiesow committed:
Merge pull request #831 from dennis531/fix-filter-size

Fix height of filters

lkiesow committed:
Merge pull request #814 from Arnei/wrong-video-in-preview-player

Fix video preview displaying wrong video

lkiesow committed:
Media type specific player

This patch changes the type of player for media assets based on their
type so we actually have player that work with the specific types.

- if `media.has_video` → `<video …`
- else if `media.has_audio` → `<audio …`
- else hide player section

This basically implements my suggestion from #814:
https://github.com/opencast/opencast-admin-interface/pull/814#discussion_r1671210608

lkiesow committed:
Simplify Opencast version display

Instead of using the Opencast bundle version directly which would be,
for example, 16.2.0 for Opencast 16.2 or 15.0.0-SNAPSHOT for a build
from the release branch, this patch turn the version into what users
actually use. That means, people need less inside information like that
17.0.0 does not mean 17.0 if there is a `SNAPSHOT` suffix.

This will now convert:
- `17.0.0-SNAPSHOT` to `17.x`
- `17.8.0` to `17.8`

lkiesow committed:
Merge pull request #817 from dennis531/event-status

Show workflow operations when clicking on status in events table

lkiesow committed:
Drop translation of old editor

This patch drops the translation of the editor from the old admin
interface which no longer exists.

lkiesow committed:
Typing of publication cell

Just removing a type exception.

Commit:
Automatically update translation keys

pselsa committed:
auto select workflow when there is only one

pselsa committed:
also do it on start task

pselsa committed:
fixed some issues

gregorydlogan committed:
Merge branch 'tabs' of JulianKniephoff/opencast-admin-interface into main

Pull request #749

  Add a very restricted form of `no-tabs` to the ESLint config

gregorydlogan committed:
Merge branch 'missing-key' of dennis531/opencast-admin-interface into main

Pull request #828

  Fix missing key in metadata translations

gregorydlogan committed:
Merge branch 'infinite-warning-messages' of dennis531/opencast-admin-interface into main

Pull request #835

  Fix recursion in effect when fetching series acls

gregorydlogan committed:
Merge branch 'player-type' of lkiesow/opencast-admin-interface into main

Pull request #842

  Media type specific player

gregorydlogan committed:
Merge branch 'simplify-version' of lkiesow/opencast-admin-interface into main

Pull request #844

  Simplify Opencast version display

gregorydlogan committed:
Merge branch 'drop-old-editor-i18n' of lkiesow/opencast-admin-interface into main

Pull request #845

  Drop translation of old editor

gregorydlogan committed:
Merge branch 'pub-cell-typing' of lkiesow/opencast-admin-interface into main

Pull request #847

  Typing of publication cell

gregorydlogan committed:
Merge branch '849-auto-select-workflow' of schuettloeffel-elsa/opencast-admin-interface into main

Pull request #850
Closes #849
  auto select workflow when there is only one

gregorydlogan committed:
Merge branch 'embed-codes' of lkiesow/opencast-admin-interface into main

Pull request #753

  Improve embed codes

lkiesow commented:
Rebased to fix resolve conflicts.

> The validation for the main metadata is not happening anymore.

Fixed.

Arnei commented:
Can confirm, validation is happening again.

But the "Previous" button on the metadata tab does nothing again. Seems to only occur if I want to schedule an event.
Here is a video of me clicking around:
[Bildschirmaufzeichnung vom 2024-07-26, 14-48-04.webm](https://github.com/user-attachments/assets/e3f03fd2-384f-4f2a-bab7-138555683725)
